linera-sdk/src/contract/conversions_to_wit.rs

```diff
@@ -78,46 +79,67 @@ impl From<log::Level> for system::LogLevel {
     }
 }
 
-impl From<ApplicationCallResult> for wit_types::ApplicationCallResult {
-    fn from(result: ApplicationCallResult) -> Self {
+impl<Effect: Serialize + DeserializeOwned + Debug, Value, Session> From<ApplicationCallResult<Effect, Value, Session>>
+    for wit_types::ApplicationCallResult
+where
+    Effect: Serialize,
+    Value: Serialize,
+    Session: Serialize,
+{
+    fn from(result: ApplicationCallResult<Effect, Value, Session>) -> Self {
         let create_sessions = result
             .create_sessions
             .into_iter()
             .map(wit_types::Session::from)
             .collect();
 
+        let value = result
+            .value
+            .map(|v| bcs::to_bytes(&v).expect("TODO"))
```

Comment:
What's the plan for the serialization error? `TryFrom` or panic?

Reply:
If `bcs` serde is symmetric, i.e. if serialization for a given struct doesn't fail it is guaranteed that deserialization won't fail, and vice versa, we could panic since our runtime is soleley responsible for performing the serde?

@jvff any thoughts on this?

Reply:
I think we should probably list the BCS serialization error scenarios, just to be sure. Can an app developer reach any of them? If that's the case, `panic`s will block the application, which could be pretty bad.

If we can guarantee that there are no failure scenarios that an app developer can reach, I agree that we should panic.

I can start looking into this :+1: